macos/FluentUI/Button/Button.swift

```diff
@@ -140,6 +148,18 @@ open class Button: NSButton {
 		}
 	}
 
+	public func addImage(forButtonState: ButtonState, image: NSImage) {
+		imagesForButtonState[forButtonState] = image
+	}
+
```

Comment:
We should have clear documentation to distinguish the difference between this method and the public property `image`. For example, does setting `image` set it for all states? or just rest? I think I would personally prefer just having one property instead of a property and a method. Something like:

```swift
exampleButton.imageForState = [
    .rest: restImage,
    .pressed: disabledImage
    .disabled: disabledImage
]
```
or just removing the image property altogether, and having the `addImage()` method take an optional parameter for state, that defaults to setting it for all the states

```swift
exampleButton1.addImage(imageForAllStates)

exampleButton2.addImage(restImage, state: .rest)
```

These are both just ideas though! Whatever is most ergonomic / easy to understand on the client side. I might go see how other Fluent buttons / frameworks have solved this as well.

Reply:
Also, to use this method in objective c, I think you need to addd the `@objc` decorator to the method.

Reply:
I think it should be closer to iOS https://developer.apple.com/documentation/uikit/uibutton/1623997-setimage

Reply:
Ah, the iOS method does exactly what I was about to propose thinking about it more (using the rest/normal state as the default if other states aren't set). Thanks @harrieshin !

Reply:
So in iOS the ImageView property is readOnly, and user is suppose to use setImage:ForState: and set it for at least   UIControlStateNormal (which can be equivalent of our .rest case). Are you guys (FX) okay with the button being alligned with IOS more than Mac in this case. (I will make the image property be readOnly then)